File: advancedscreenswitchersample/src/main/java/me/aartikov/advancedscreenswitchersample/ui/TabFragment.java
package me.aartikov.advancedscreenswitchersample.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import me.aartikov.advancedscreenswitchersample.R;
import me.aartikov.advancedscreenswitchersample.SampleApplication;
import me.aartikov.advancedscreenswitchersample.screens.InnerScreen;
import me.aartikov.advancedscreenswitchersample.screens.TabScreen;
import me.aartikov.alligator.Navigator;
import me.aartikov.alligator.annotations.RegisterScreen;


@RegisterScreen(TabScreen.class)
public class TabFragment extends Fragment implements ContainerIdProvider {

	TextView mNameTextView;

	private final Navigator mNavigator = SampleApplication.getNavigator();

	@Override
	public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
		return inflater.inflate(R.layout.fragment_tab, container, false);
	}

	@Override
	public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mNameTextView = view.findViewById(R.id.name_text_view);

		TabScreen screen = SampleApplication.getScreenResolver().getScreen(this);
		mNameTextView.setText(screen.getName());
	}

	@Override
	public void onResume() {
		super.onResume();
		setInitialFragmentIfRequired();
	}

	private void setInitialFragmentIfRequired() {
		boolean hasFragment = getChildFragmentManager().findFragmentById(R.id.inner_container) != null;
		if (!hasFragment && mNavigator.canExecuteCommandImmediately()) {
			mNavigator.reset(new InnerScreen(1));
		}
	}

	@Override
	public int getContainerId() {
		return R.id.inner_container;
	}
}

File: advancedscreenswitchersample/src/main/java/me/aartikov/advancedscreenswitchersample/ui/TabFragment.kt
package me.aartikov.advancedscreenswitchersample.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import me.aartikov.advancedscreenswitchersample.R
import me.aartikov.advancedscreenswitchersample.SampleApplication
import me.aartikov.advancedscreenswitchersample.screens.InnerScreen
import me.aartikov.advancedscreenswitchersample.screens.TabScreen
import me.aartikov.alligator.Navigator
import me.aartikov.alligator.annotations.RegisterScreen

@RegisterScreen(TabScreen::class)
class TabFragment : Fragment(), ContainerIdProvider {

    private lateinit var mNameTextView: TextView

    private val mNavigator: Navigator = SampleApplication.navigator

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_tab, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mNameTextView = view.findViewById(R.id.name_text_view)

        val screen = SampleApplication.screenResolver.getScreen<TabScreen>(this)
        mNameTextView.text = screen.name
    }

    override fun onResume() {
        super.onResume()
        setInitialFragmentIfRequired()
    }

    private fun setInitialFragmentIfRequired() {
        val hasFragment = childFragmentManager.findFragmentById(R.id.inner_container) != null
        if (!hasFragment && mNavigator.canExecuteCommandImmediately()) {
            mNavigator.reset(InnerScreen(1))
        }
    }

    override val containerId: Int
        get() = R.id.inner_container
}
